src/Foundation.Cms/SchemaMarkup/RegisterSchemaMappers.cs

```diff
@@ -0,0 +1,33 @@
+﻿using EPiServer.Framework;
+using EPiServer.Framework.Initialization;
+using EPiServer.ServiceLocation;
+
+namespace Foundation.Cms.SchemaMarkup
+{
+    /// <summary>
+    /// Module to scan assemblies on initialisation for instances of ISchemaDataMapper and register with StructureMap
+    /// </summary>
+    [InitializableModule]
+    [ModuleDependency(typeof(EPiServer.Web.InitializationModule))]
+    public class RegisterSchemaMappers : IConfigurableModule
+    {
+        public void ConfigureContainer(ServiceConfigurationContext context)
+        {
+            context.StructureMap().Configure(x =>
+            {
```

Comment:
Yep, can do. I was expecting that to be the controversial bit as I know a fair few people have a preference for explicit registration rather than scanning. The intention was really to simplify the process of adding a mapper and to allow features to be more self contained. Explicitly registering the mappers places a reliance on a single file where all explicit registrations are performed, making it slightly more difficult to remove features (as that file will typically be outside of the feature but have a dependency on it). That said, I understand structuremap is an option in an Episerver site rather than a firm dependency so it's best to avoid referencing it directly.

Reply:
Right, I've removed that file and added in explicit registration of the mappers in each project's initialisation file.